python311Packages.weatherflow4py: init at 0.2.20

fabaff opened the pull request:
Module to interact with the WeatherFlow REST API

https://github.com/jeeftor/weatherflow4py

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabaff commented:
Waiting on a response for the license statement: https://github.com/jeeftor/weatherflow4py/issues/24

fabaff commented:
Result of `nixpkgs-review pr 294274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.weatherflow4py</li>
    <li>python311Packages.weatherflow4py.dist</li>
    <li>python312Packages.weatherflow4py</li>
    <li>python312Packages.weatherflow4py.dist</li>
  </ul>
</details>

fabaff commented:
Result of `nixpkgs-review pr 294274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.weatherflow_cloud</li>
    <li>python311Packages.weatherflow4py</li>
    <li>python311Packages.weatherflow4py.dist</li>
    <li>python312Packages.weatherflow4py</li>
    <li>python312Packages.weatherflow4py.dist</li>
  </ul>
</details>

mweinelt reviewed (dismissed):
Please squash init, format, update commits into a single commit. There is no value to having these separate.

Frankly, this is precisely what I mean when I talk about churn, that made me do the treewide format.

fabaff commented:
Result of `nixpkgs-review pr 294274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.weatherflow_cloud</li>
    <li>python311Packages.weatherflow4py</li>
    <li>python311Packages.weatherflow4py.dist</li>
    <li>python312Packages.weatherflow4py</li>
    <li>python312Packages.weatherflow4py.dist</li>
  </ul>
</details>

mweinelt commented on pkgs/development/python-modules/weatherflow4py/default.nix:
```suggestion
  build-system = [ poetry-core ];
```

mweinelt commented on pkgs/development/python-modules/weatherflow4py/default.nix:
```suggestion
  dependencies = [
```

mweinelt reviewed (commented):
Two nits, but we can also address them in bulk at some point. Feel free to merge with or without them.

fabaff committed:
python311Packages.weatherflow4py: init at 0.2.20

Module to interact with the WeatherFlow REST API

https://github.com/jeeftor/weatherflow4py

fabaff committed:
home-assistant: update component-packages

fabaff commented:
Result of `nixpkgs-review pr 294274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.weatherflow_cloud</li>
    <li>python311Packages.weatherflow4py</li>
    <li>python311Packages.weatherflow4py.dist</li>
    <li>python312Packages.weatherflow4py</li>
    <li>python312Packages.weatherflow4py.dist</li>
  </ul>
</details>
